File: examples/search-algorithms/metrics.ts
export function extract(result: {
	comparisons?: number;
	found?: boolean;
	executionTimeMs?: number;
}): Record<string, number> {
	return {
		comparisons: result.comparisons ?? 0,
		found: result.found ? 1 : 0,
		executionTimeMs: result.executionTimeMs ?? 0,
	};
}

File: examples/search-algorithms/metrics.py
"""Search Algorithms Metrics Extractor."""


def extract(result):
    r = result if isinstance(result, dict) else {}
    return {
        "comparisons": r.get("comparisons", 0),
        "found": 1 if r.get("found") else 0,
        "executionTimeMs": r.get("executionTimeMs", 0),
    }
